File: Week_01/G20190343020242/LeetCode_283_0242.js
/*
 * @lc app=leetcode.cn id=283 lang=javascript
 *
 * [283] 移动零
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let j = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] != 0) {
            temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
            j ++
        }
    }
};
// @lc code=end

File: Week_01/G20190343020242/LeetCode_283_0242.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start


class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        j = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                # nums[j] = nums[i]
                # if i != j:
                #     nums[i] = 0
                nums[j], nums[i] = nums[i], nums[j]
                j += 1

# @lc code=end
